.github/workflows/build-firehose-version.yml

```diff
@@ -0,0 +1,103 @@
+name: Build firehose version
+run-name: ${{ inputs.dryrun && '[dryrun] ' || ''}}Creating p-${{ inputs.tag }}-fh <> cosmos/${{ inputs.tag }}..graphprotocol/${{inputs.fh-tag}}
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        type: string
+        required: true
+        description: 'cosmos/cosmos-sdk tag name'
+      fh-tag:
+        type: string
+        required: true
+        description: 'graphprotocol/cosmos-sdk firehose tag name'
+      dryrun:
+        type: boolean
+        required: false
+        default: false
+
+env:
+  p_tag: p-${{ inputs.tag }}-fh
+
+jobs:
+  cosmos-sdk-fh:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.WORKFLOW_TOKEN }}
```

Comment:
no required, checkout will work with current branch properly i feel

Reply:
Yes, checkout works with default token.
This `WORKFLOW_TOKEN` is required when newly created tag modifies anything in `.github/workflows` directory (which requires `workflow` permission, and the default token does not have it)

One solution as you suggested is to ignore the changes under `.github/workflows` directory. And other is create the token.
To create the token we need a PAT (Personal Access Token) with `workflow` permission. And then under repo's `/settings/secrets/action` we can add a token named `WORKFLOW_TOKEN` with the PAT value.

Reply:
hmm i dont seem to have the permissions to create a PAT for the org. 
Also we are not able to create a PAT specific to repo, but to the user itself

Reply:
Can we try and ignore the workflows?

Reply:
Yes, ignoring workflows directory before pushing works with default token. So there will be one extra commit (for deleting `.github/workflows` dir) with every tag we create.

I'm adding a build step in between as well.

Reply:
Also Pranjal changed the main branch to `main-wf`. So I've changed the base branch of this PR.
Reason being that with workflow being in the main branch we can trigger this manually via Actions UI, otherwise we had to do this via `gh` command line only.